flink-ml-lib/src/test/java/org/apache/flink/ml/regression/LinearRegressionTest.java

```diff
@@ -0,0 +1,240 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+/** Tests {@link LinearRegression} and {@link LinearRegressionModel}. */
+public class LinearRegressionTest {
+
+    @Rule public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private StreamExecutionEnvironment env;
+
+    private StreamTableEnvironment tEnv;
+
+    private static final List<Row> trainData =
+            Arrays.asList(
+                    Row.of(Vectors.dense(2, 1), 4.0, 1.0),
+                    Row.of(Vectors.dense(3, 2), 7.0, 1.0),
+                    Row.of(Vectors.dense(4, 3), 10.0, 1.0),
+                    Row.of(Vectors.dense(2, 4), 10.0, 1.0),
+                    Row.of(Vectors.dense(2, 2), 6.0, 1.0),
+                    Row.of(Vectors.dense(4, 3), 10.0, 1.0),
+                    Row.of(Vectors.dense(1, 2), 5.0, 1.0),
+                    Row.of(Vectors.dense(5, 3), 11.0, 1.0));
+
+    private static final double[] expectedCoefficient = new double[] {1.0, 2.0};
+
+    private static final double TOLERANCE = 1e-7;
+
+    private Table trainDataTable;
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.setParallelism(4);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+        Collections.shuffle(trainData);
+        trainDataTable =
+                tEnv.fromDataStream(
+                        env.fromCollection(
+                                trainData,
+                                new RowTypeInfo(
+                                        new TypeInformation[] {
+                                            TypeInformation.of(DenseVector.class),
+                                            Types.DOUBLE,
+                                            Types.DOUBLE
+                                        },
+                                        new String[] {"features", "label", "weight"})));
+    }
+
+    @SuppressWarnings("unchecked")
+    private void verifyPredictionResult(
+            Table output, String labelCol, String weightCol, String predictionCol)
+            throws Exception {
+        List<Row> predResult = IteratorUtils.toList(tEnv.toDataStream(output).executeAndCollect());
+        double lossSum = 0;
+        for (Row predictionRow : predResult) {
+            double label = (double) predictionRow.getField(labelCol);
+            double prediction = (double) predictionRow.getField(predictionCol);
+            double weight = (double) predictionRow.getField(weightCol);
+            lossSum += weight * Math.pow(label - prediction, 2);
+        }
+        assertTrue(lossSum < 1.0);
```

Comment:
I'm not sure whether this criteria can guarantee the correctness of this algorithm. Is it possible to provide an estimation of the expected prediction result, and tests that the actual prediction should be close enough to the expected value, for example using `expectedCoefficient`?

Reply:
It is a regression task, so the prediction is NOT a label. It is not possible to assert that each prediction is correct as a label.

I have refined the function and assert that the average loss is smaller than 0.1.

Reply:
In `testGetModelData` it has been possible to verify the coefficient value, so I suppose the prediction result computed from the coefficient and the input data should also be predictable.

Reply:
Sounds good. I have updated the PR.